context_test.go

```diff
@@ -0,0 +1,39 @@
+package vesper
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPayloadFromContext(t *testing.T) {
+	in := []byte(`
+{
+	"message": "hello world"
+}
+`)
+	called := false
+	v := New(func(ctx context.Context, i interface{}) (int, error) {
+		payload, ok := PayloadFromContext(ctx)
+		assert.True(t, ok)
+		assert.Equal(t, in, payload)
+		called = true
+		return 1, nil
+	})
+	_, _ = v.handler.Invoke(context.Background(), in)
+	assert.True(t, called)
+}
+
+func TestHandlerSignatureFromContext(t *testing.T) {
+	called := false
+	v := New(func(ctx context.Context, i interface{}) error {
+		sig, ok := HandlerSignatureFromContext(ctx)
+		assert.True(t, ok)
+		assert.Equal(t, "func(context.Context, interface {}) error", sig.String())
```

Comment:
I assume we can still get the second argument to this signature if we want?

Reply:
Yep, I changed it to show that you can get a typed argument

Reply:
Ha, sorry I meant rather than getting the string `vesper.user` you would get the actual arguments `reflect.Type` 

All good, I was being lazy (on a call), i'll test locally myself. Let's get this in now though.

Reply:
Oh yes! :D `sig.In(1)` gives you back the `reflect.Type` of `vesper.User`